packages/frontend/templates/build-edgeless.mjs

```diff
@@ -0,0 +1,289 @@
+        assetsFiles.push(fileObj);
+        return;
+      }
+
+      if (name.endsWith('.snapshot.json')) {
+        snapshotFiles.push(fileObj);
+        return;
+      }
+    });
+
+    await Promise.all(
+      assetsFiles.map(async file => {
+        const blob = await file.async('blob');
+        const arrayBuffer = await blob.arrayBuffer();
+        const buffer = Buffer.from(arrayBuffer, 'binary');
+
+        await promisify(fs.writeFile)(
+          join(ASSETS_PATH, file.name.replace('assets/', '')),
+          buffer
+        );
+      })
+    );
+
+    await Promise.all(
+      snapshotFiles.map(async snapshot => {
+        const json = await snapshot.async('text');
+        const snapshotContent = JSON.parse(json);
+        let previewContent = '';
+
+        if (fs.existsSync(join(ZIP_PATH, `${templateName}.svg`))) {
+          const previewFile = fs.readFileSync(
+            join(ZIP_PATH, `${templateName}.svg`),
+            'utf-8'
+          );
+          previewContent = previewFile
+            .replace(/\n/g, '')
+            .replace(/\s+/g, ' ')
+            .replace('fill="white"', 'fill="currentColor"');
+        } else {
+          console.warn(`No preview found for ${templateName}`);
+        }
+
+        convertSourceId(snapshotContent.blocks);
+
+        const template = {
+          name: templateName,
+          type: 'template',
+          preview: previewContent,
+          content: snapshotContent,
+        };
+
+        await promisify(fs.writeFile)(
+          join(join(TEMPLATE_PATH, `${templateName}.json`)),
+          JSON.stringify(template, undefined, 2)
+        );
+
+        templates.push(templateName);
+      })
+    );
+  }
+
+  return templates;
+};
+
+function numberToWords(n) {
+  const ones = [
+    'Zero',
+    'One',
+    'Two',
+    'Three',
+    'Four',
+    'Five',
+    'Six',
+    'Seven',
+    'Eight',
+    'Nine',
+  ];
+
+  if (n < 10) {
+    return ones[n];
+  } else {
+    throw new Error(`Not implemented: ${n}`);
+  }
+}
+
+const camelCaseNumber = variable => {
+  const words = variable.split(' ');
+  return words
+    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+    .join('');
+};
+
+const toVariableName = name => {
+  const converted = Array.from(name).reduce((pre, char) => {
+    if (char >= '0' && char <= '9') {
+      return pre + numberToWords(char - '0');
+    }
+
+    return pre + char;
+  }, '');
+
+  return camelCaseNumber(converted);
+};
+
+/**
+ *
+ * @param {Array<string>} templates
+ */
+const buildScript = async templates => {
+  const templateVariableMap = templates.reduce((map, template) => {
+    map[template] = toVariableName(template);
+    return map;
+  }, {});
+  const importStatements = templates
+    .map(template => {
+      return `import ${toVariableName(
+        template
+      )} from './edgeless/${template}.json';`;
+    })
+    .join('\n');
+
+  const code = `${importStatements}
+
+const templates = {
```

Comment:
can you reorg the files stored in edgeless-snapshots in a pattern and then enumerate the template meta, instead of maintain these object explicity? I think in the long term what we want is to replace snapshot files only without touching this script

Reply:
Good idea.